### PR TITLE
Ticket #1998: Add table descriptions

### LINKED ISSUE #1998: Django admin: Add table descriptions
### Issue description

We need to add basic descriptions for each table in the admin. This will help new users as well as established users who are unfamiliar with a specific table.

- Descriptions should appear above each table, directly below the page heading. 
  - See mock-up below, but note that it's simply a modified screenshot. Padding should be added so the description has some breathing room above/below it.
- Show approx 150 characters, followed by an ellipses with a "show more" link to read the rest of the text.
  - May need to adjust that length once design reviews in the UI
- When full text is displayed, the "show more" button should change to "show less" and appear at the end of the entry.
- Some descriptions will contain links to other tables in the admin.

### Acceptance criteria

- [ ] Each table in the admin displays the corresponding description found in this document: [Table descriptions](https://docs.google.com/document/d/1eu92s38akKP-zHVUiVYi79HGezk4AccubGiDhAATuYk/edit?pli=1#heading=h.b89opg8sn758)
- [ ] During implementation, work with design to confirm details on how it should display

NOTE: "Federal agencies" does not have a description yet because that feature is not fully implemented.

### Additional context

Reach out to Katherine with any questions about the descriptions.

### Links to other issues

Content ticket: #1040

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #1998

## Changes

- Adds a show/hide script in get-gov-admin.js
- Custom styles
- Added model_descriptions.html for each table description
- Modifies the base change_list.html file to include the new file, model_descriptions.html
- Adds unit tests

## Context for reviewers

This PR adds a custom description for most tables in Django admin. For fields that do not have a description as defined in this document, the table will have the description "This table does not have a description yet." which serves as a default for any new tables we may add in the future

## Setup
As an analyst or a superuser, go to each model (such as Domain Information or Domain Request) and verify that the content looks good.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots
image

### CODE REVIEW

File: src/registrar/templates/admin/model_descriptions.html [hunk 1]
```diff
@@ -0,0 +1,204 @@
+
+<div class="dja__model-description text-normal">
+    {% if opts.model_name == 'domainrequest' %}
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/templates/admin/model_descriptions.html:3]: Put them in includes in a descriptions folder?

File: src/registrar/tests/test_admin.py [hunk 2]
```diff
@@ -852,6 +878,23 @@ def setUp(self):
         )
         self.mock_client = MockSESClient()
 
+    @less_console_noise_decorator
+    def test_has_model_description(self):
```
Comment (rachidatecs) [inline src/registrar/tests/test_admin.py:882]: I appreciate that you didn't try to make a helper for this test

**Review by rachidatecs — COMMENTED** [review]
Code looks good. Look at my suggestion.

I'll run this later and finish the review.

File: src/registrar/assets/sass/_theme/_admin.scss [hunk 3]
```diff
@@ -659,3 +659,37 @@ form .aligned p.help, form .aligned div.help {
...
+
+    @media (prefers-color-scheme: light) {
+        p, li {
+            color: var(--body-fg);
+        }
+    }
+
+    @media (prefers-color-scheme: dark) {
+        p, li {
+            color: var(--primary);
```
Comment (rachidatecs) [inline src/registrar/assets/sass/_theme/_admin.scss:682]: body-bg

File: src/registrar/assets/sass/_theme/_admin.scss [hunk 4]
```diff
@@ -659,3 +659,37 @@ form .aligned p.help, form .aligned div.help {
...
+        }
+    }
+
+    li {
+        list-style-type: disc;
+    }
+
+    &.dja__model-description--no-overflow {
+        display: block;
+        overflow: unset;
```
Comment (rachidatecs) [inline src/registrar/assets/sass/_theme/_admin.scss:692]: auto instead of unset

**Review by Katherine-Osos — APPROVED** [review]
LGTM!

File: src/registrar/templates/django/admin/includes/descriptions/domain_invitation_description.html [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+<p>
+An “invited” status indicates that the recipient has not logged in to the registrar since the invitation was sent. 
+A “received” status indicates that the recipient has logged in.
+</p>
+
+<p>
+If an invitation is created in this table, an email will not be sent. 
+To have an email sent, go to the domain in <a class="usa-button usa-button--unstyled text-underline" href="{% url 'admin:registrar_domain_changelist' %}">Domains</a>, 
+click the “Manage domain” button, and add a domain manager. 
+</p>
```
Comment (rachidatecs) [inline src/registrar/templates/django/admin/includes/descriptions/domain_invitation_description.html:16]: Nitpick: Alysia like the trailing lines

**Review by rachidatecs — COMMENTED** [review]
Love the file reorg.

The media query for dark mode is messing up light mode. I think it's going off the browser or OS setting, not DJA. What you had before worked better, just use the grays not blues.

Screenshot 2024-05-03 at 1 27 58 PM

You'll need to handle the link color (screenshot taken in OS dark mode, DJA light mode)
Screenshot 2024-05-03 at 1 28 04 PM

Overall, suggest testing this in OS light mode + DJA light/dark mode, OS dark mode + DJA light/dark mode in a Chromium browser and in FF (where I had some weird issues that required specific overrides on other tickets)

**Review by rachidatecs — APPROVED** [review]
Approved with a suggested follow-up to clean up our various show/hide buttons @abroddrick (refactor or account for it as part of the next show/hide ticket)

**Pull request merged**